app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -10,27 +10,45 @@
 use Illuminate\Queue\InteractsWithQueue;
 use Illuminate\Contracts\Queue\ShouldQueue;
 use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Support\Str;
 
 class RockTheVoteRecord {
     public function __construct($record)
     {
-        $this->post_type = 'voter-reg';
-        $this->source = 'rock-the-vote';
-
+        $this->addr_street1 = $record['Home address'];
+        $this->addr_street2 = $record['Home unit'];
+        $this->addr_city = $record['Home city'];
+        $this->addr_state = $record['Home state'];
+        $this->addr_zip = $record['Home zip code'];
         $this->email = $record['Email address'];
+        $this->first_name = $record['First name'];
+        $this->last_name = $record['Last name'];
         $this->mobile = $record['Phone'];
-        $this->email_opt_in = $record['Opt-in to Partner email?'];
+
+        $emailOptIn = $record['Opt-in to Partner email?'];
+        if ($emailOptIn) {
+            $this->email_subscription_status = str_to_boolean($emailOptIn);
+        }
         // Note: Not a typo, this column name does not have the trailing question mark.
-        $this->sms_opt_in = $record['Opt-in to Partner SMS/robocall'];
+        $smsOptIn = $record['Opt-in to Partner SMS/robocall'];
+        if ($smsOptIn && $this->mobile) {
+            $this->sms_status = str_to_boolean($smsOptIn) ? 'active' : 'stop';
+        }
 
-        $this->voter_registration_status = $this->parseVoterRegistrationStatus($record['Status'], $record['Finish with State']);
+        $rtvStatus = $this->parseVoterRegistrationStatus($record['Status'], $record['Finish with State']);
+       
+        $this->voter_registration_status =  Str::contains($rtvStatus, 'register') ? 'registration_complete' : $rtvStatus;
 
         $referral = $this->parseReferralCode($record['Tracking Source']);
-        $this->user_id = $referral['northstar_id'];
-        $this->campaign_id = (int) $referral['campaign_id'];
-        $this->source_details = $referral['source_details'];
-
-        // TODO: Add all other properties used for getting/validating/creating users/posts.
+        $this->user_id = !empty($referral['user_id']) ? $referral['user_id'] : null;
+        // TODO: Do we need to check the referral for these values? Don't seem they are ever used.
+        $this->campaign_id = 8017;
```

Comment:
is this hardcoding the campaign so it is `8017` each time, since it happens after `parseReferralCode`?

Reply:
It is -- it's essentially ignoring the campaign found in `parseReferralCode`. I added this in, since I wasn't sure if the desired behavior was to create a post for 8017, or for the campaign found in a referral code... which is related to thread in https://github.com/DoSomething/chompy/pull/60#discussion_r259963459

Reply:
I might have been misunderstanding your Q the whole time! Are you asking for the context for why we want to default to `8017` each time instead of getting what is in `parseReferralCode`?

Reply:
I was! @ngjo and I chatted through the history of these imports today, and landed on the requirements we need for present day referral codes:

* When fetching a user,  use the parsed `user` referral parameter only when the parsed `referral` referral parameter exists and is set to `true` (otherwise, check for user by email, then mobile)

* Ignore the `campaign_id`, `campaign_run_id`, `source`, and `source_details` referral parameters completely.

    * When creating the post, send an action ID which will be set via hardcoding or config var -- once https://github.com/DoSomething/rogue/pull/837 is deployed, refs https://github.com/DoSomething/chompy/pull/58/files#r258541121 and [Slack thread](https://dosomething.slack.com/archives/CA0N5FXND/p1551114281011600)

    * Continue to send hardcoded post source as `rock-the-vote`, `source_details` as null

* We will continue to send the post `details` as is (renamed as `parsePostDetails` in this PR) 

I'll update this PR description and ping folks when this ready for 👀 . Thanks for taking a look @chloealee !

Reply:
This is a great summary and so much clearer than what we have as-is! Thanks for taking the time to get to the bottom of this and thanks for @ngjo for clearing up! My memory is so bad! 👵 Will keep an eye out for your notification when it is ready for another look!